services/gitdiff/highlightdiff_test.go

```diff
@@ -20,11 +19,13 @@ func TestDiffWithHighlight(t *testing.T) {
 		"		run(db)\n",
 	)
 
-	expected := `		<span class="n">run</span><span class="o">(</span><span class="removed-code"><span class="k">&#39;</span><span class="o">&lt;</span><span class="o">&gt;</span><span class="k">&#39;</span></span><span class="o">)</span>`
+	expected := `		run(<span class="removed-code">'<>'</span>)
```

Comment:
It can't be right. By your approach, there will be XSS. Something like `<script>xxx</script>` will be rendered to the user's page.

Reply:
Hey, thanks for reviewing.
I didn't change how html is encoded.

`<span class="removed-code">'<>'</span>` should render something like  `'<>'` to the page.

If I'm misunderstanding something, my apologies.


![Screenshot from 2025-03-02 00-33-03](https://github.com/user-attachments/assets/2227a0c3-fc43-449c-bac8-e7aa404ccd40)
![image](https://github.com/user-attachments/assets/7fd6e5a8-ced9-445b-8a43-ae871ac3aa5c)

Reply:
In your test result:

```
"run(<span class="removed-code">'<>'</span>)"
```

There are 3 `<` chars. The first one is `<span` (added by the render), the second one is the one in `'<>'`

The old logic does encode the second one to `&lt;`, which is necessary to render it to browser.

But in you case, the second one is still `<`, then it will be used as "HTML tag", but not rendered as "code content"

I do not see why it could be right.

Reply:
oh, it's because it's no longer calling `highlight.Code(filename, language, codeA)` here. So, it's not actually "rendering" the html here.

Reply:
It's not really related to "rendering" or not. Let's take a simpler example below:

You made this output:

```
`a <span class="removed-code"><</span> b`
```

What "kind" of the string it is?

* If it is a HTML string, then it is invalid, the `<` shouldn't appear in HTML content. It must be encoded to `&lt;`
* If it is not a HTML string, then `<span ...>` wouldn't be rendered as HTML but just be displayed as-is, actually it is not the case because this tag is for highlighting the code as removed.

I do not see it could be right.

Reply:
If you think your approach is right but the tests do not work for your approach, please design correct tests, and use the existing test data to verify that the logic overall is right.